File: 0019-Remove-Nth-Node-From-End-of-List/lc0019.py
# Python solution.

# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def removeNthFromEnd(self, head: ListNode, n: int) -> ListNode:
        dumb = ListNode(0)
        dumb.next = head
        p1, p2 = dumb, dumb
        for i in range(n+1):
            p2 = p2.next
        while p2 != None:
            p1 = p1.next
            p2 = p2.next
        p1.next = p1.next.next
        return dumb.next

File: 0019-Remove-Nth-Node-From-End-of-List/lc0019.js
// JavaScript solution.

/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} n
 * @return {ListNode}
 */

var removeNthFromEnd = function(head, n) {
    let dumb = new ListNode(0);
    dumb.next = head;
    let p1 = dumb;
    let p2 = dumb;
    for (let i = 0; i < n+1; i++) {
        p2 = p2.next;
    }
    while (p2 !== null) {
        p1 = p1.next;
        p2 = p2.next;
    }
    p1.next = p1.next.next;
    return dumb.next;
};
